zerver/lib/message.py

```diff
@@ -538,9 +538,9 @@ def access_message(user_profile: UserProfile, message_id: int) -> Tuple[Message,
 
 def has_message_access(user_profile: UserProfile, message: Message,
                        user_message: Optional[UserMessage]) -> bool:
-    if user_message is None:
+    if user_message is None and message.sender != user_profile:
```

Comment:
Maybe it'd be cleaner to do `if message.sender_id == user_profile.id: return True` at the top as a separate conditional?  It'd make test coverage verify we've tested that code path.

(Might make sense to swap things to have `if user_message is not None: return True` as well, for similar readability/testing reasons)